lib/rubocop/cop/style/legacy_hash.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks the usage of pre-2.4 `Hash[args]` method of converting arrays to hashes
+      #
+      # @example
+      #   # bad
+      #   Hash[ary]
+      #
+      #   # good
+      #   ary.to_h
+      #
+      #   # bad
+      #   Hash[key1, value1, key2, value2]
+      #
+      #   # good
+      #   {key1 => value1, key2 => value2}
+      class LegacyHash < Base
+        extend AutoCorrector
+
+        MSG_TO_H = 'Prefer ary.to_h to Hash[ary].'
+        MSG_LITERAL = 'Prefer literal hash to Hash[arg1, arg2, ...].'
+        MSG_SPLAT = 'Prefer array_of_pairs.to_h to Hash[*array].'
+        RESTRICT_ON_SEND = %i[[]].freeze
+
+        def_node_matcher :hash_from_array?, '(send (const nil? :Hash) :[] ...)'
```

Comment:
Might be useful to catch `::Hash` as well:

```suggestion
        def_node_matcher :hash_from_array?, '(send (const {nil? cbase} :Hash) :[] ...)'
```